### DISCUSSION TITLE
Issue with installing golang on wsl2 - (gzip: stdin: not in gzip format)

### DISCUSSION THREAD

**Discussion by missioin-peaks** [microsoft/WSL#6021] [General]
I am having issue installing golang using the tar file on my Ubuntu 20.04 Linux distro on WSL 2. 

This is the error I am getting:

mp@L00000:~$ tar -C /usr/local -xzvf go1.15.2.linux-amd64.tar.gz

gzip: stdin: not in gzip format
tar: Child returned status 1
tar: Error is not recoverable: exiting now

I downloaded the tar/gz file using the following wget (I also tried with curl, but faced same issue):

 mp@L00000:~$ wget https://golang.org/dl/go1.15.2.linux-amd64.tar.gz -o go1.15.2.linux-amd64.tar.gz

This is my Linux distro:

No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.1 LTS
Release:        20.04
Codename:       focal

Thanks,
Raj

**Comment by Faizan007-hub** [comment]
I am having issue installing golang using the tar file
                                      wget https://golang.org/dl/go1.20.4.linux-amd64.tar.gz
I am having issue installing golang using the tar file
                              rm -rf /usr/local/go && tar -C /usr/local -xzf go1.20.4.linux-amd64.tar.gz
tar (child): go1.20.4.linux-amd64.tar.gz: Cannot open: No such file or directory
tar (child): Error is not recoverable: exiting now
tar: Child returned status 2
tar: Error is not recoverable: exiting now

**Reply by Biswa96** [reply]
Try to download the Linux file from here https://go.dev/dl/
